source/common/uma_helpers.hpp

```diff
@@ -0,0 +1,185 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef UR_UMA_HELPERS_H
+#define UR_UMA_HELPERS_H 1
+
+#include <uma/memory_pool.h>
+#include <uma/memory_pool_ops.h>
+#include <uma/memory_provider.h>
+#include <uma/memory_provider_ops.h>
+
+#include <stdexcept>
+#include <tuple>
+#include <utility>
+
+#include <iostream>
+
+namespace uma {
+using ur_memory_provider_handle_unique =
+    std::unique_ptr<uma_memory_provider_t, decltype(&umaMemoryProviderDestroy)>;
+using ur_memory_pool_handle_unique =
+    std::unique_ptr<uma_memory_pool_t, decltype(&umaPoolDestroy)>;
+
+// C++14 version of std::make_from_tuple (but uses operator new)
```

Comment:
You can use `std::make_from_tuple` now as the current standard in UR is C++17.

Reply:
It's still problematic since I need to allocate the `T` dynamically. I could probably use something like this: `new T(std::make_from_tuple<T>(args...);` but that would only work for moveable types